File: Components/DevExpressVCL/Demos/ExpressScheduler/CBuilder/WebServiceDemo/WebServiceDemoSetupForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "WebServiceDemoSetupForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "cxMemo"
#pragma link "cxRichEdit"
#pragma link "cxTextEdit"
#pragma link "dxCustomWizardControl"
#pragma link "dxLayoutContainer"
#pragma link "dxLayoutControl"
#pragma link "dxLayoutcxEditAdapters"
#pragma link "dxWizardControl"
#pragma link "cxButtons"
#pragma link "dxLayoutControlAdapters"
#pragma link "dxLayoutLookAndFeels"
#pragma resource "*.dfm"
TWebServiceDemoSetupWizard *WebServiceDemoSetupWizard;
//---------------------------------------------------------------------------
__fastcall TWebServiceDemoSetupWizard::TWebServiceDemoSetupWizard(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TWebServiceDemoSetupWizard::FormCreate(TObject *Sender)
{
	reAbout->Lines->LoadFromFile(ExtractFilePath(Application->ExeName) + "Wizard.rtf");
	reMSGraph->Lines->LoadFromFile(ExtractFilePath(Application->ExeName) + "Wizard-MSGraph.rtf");
	reGoogleApi->Lines->LoadFromFile(ExtractFilePath(Application->ExeName) + "Wizard-GoogleAPI.rtf");
}
//---------------------------------------------------------------------------
void __fastcall TWebServiceDemoSetupWizard::reURLClick(TcxCustomRichEdit *Sender,
          const UnicodeString URLText, TMouseButton Button)
{
  	dxShellExecute(Sender->Handle, URLText);
}
//---------------------------------------------------------------------------
void __fastcall TWebServiceDemoSetupWizard::teChange(TObject *Sender)
{
	UpdateStateButtons();
}
//---------------------------------------------------------------------------
void __fastcall TWebServiceDemoSetupWizard::UpdateStateButtons()
{
  btnStart->Enabled = ((Trim(teMSGraphClientID->Text) != "") && (Trim(teMSGraphClientSecret->Text) != "")) ||
	((Trim(teGoogleApiClientID->Text) != "") && (Trim(teGoogleApiClientSecret->Text) != ""));
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressScheduler/Delphi/WebServiceDemo/WebServiceDemoSetupForm.pas
unit WebServiceDemoSetupForm;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, cxGraphics, cxControls, cxLookAndFeels, cxLookAndFeelPainters,
  cxContainer, cxEdit, cxTextEdit, cxMemo, cxRichEdit, dxCustomWizardControl,
  dxWizardControl, dxLayoutcxEditAdapters, dxLayoutContainer, cxClasses,
  dxLayoutControl, dxLayoutLookAndFeels, cxCheckBox, dxLayoutControlAdapters,
  cxButtons, Menus, StdCtrls;

type
  TWebServiceDemoSetupWizard = class(TForm)
    dxLayoutControl2: TdxLayoutControl;
    reGoogleApi: TcxRichEdit;
    teGoogleApiClientSecret: TcxTextEdit;
    teGoogleApiClientID: TcxTextEdit;
    dxLayoutGroup1: TdxLayoutGroup;
    dxLayoutItem4: TdxLayoutItem;
    dxLayoutItem5: TdxLayoutItem;
    dxLayoutItem6: TdxLayoutItem;
    dxLayoutLookAndFeelList1: TdxLayoutLookAndFeelList;
    dxLayoutCxLookAndFeel1: TdxLayoutCxLookAndFeel;
    dxLayoutGroup2: TdxLayoutGroup;
    dxLayoutGroup3: TdxLayoutGroup;
    dxLayoutItem1: TdxLayoutItem;
    reMSGraph: TcxRichEdit;
    dxLayoutItem2: TdxLayoutItem;
    teMSGraphClientID: TcxTextEdit;
    dxLayoutItem3: TdxLayoutItem;
    teMSGraphClientSecret: TcxTextEdit;
    dxLayoutGroup4: TdxLayoutGroup;
    dxLayoutGroup5: TdxLayoutGroup;
    btnStart: TcxButton;
    dxLayoutItem7: TdxLayoutItem;
    btnCancel: TcxButton;
    dxLayoutItem9: TdxLayoutItem;
    dxLayoutItem10: TdxLayoutItem;
    reAbout: TcxRichEdit;
    procedure teChange(Sender: TObject);
    procedure reURLClick(Sender: TcxCustomRichEdit;
      const URLText: string; Button: TMouseButton);
    procedure FormCreate(Sender: TObject);
  private
    procedure UpdateStateButtons;
  public
    { Public declarations }
  end;

var
  WebServiceDemoSetupWizard: TWebServiceDemoSetupWizard;

implementation

uses
  dxCore;

{$R *.dfm}

procedure TWebServiceDemoSetupWizard.FormCreate(Sender: TObject);
begin
  reAbout.Lines.LoadFromFile(ExtractFilePath(Application.ExeName) + 'Wizard.rtf');
  reMSGraph.Lines.LoadFromFile(ExtractFilePath(Application.ExeName) + 'Wizard-MSGraph.rtf');
  reGoogleApi.Lines.LoadFromFile(ExtractFilePath(Application.ExeName) + 'Wizard-GoogleAPI.rtf');
end;

procedure TWebServiceDemoSetupWizard.reURLClick(
  Sender: TcxCustomRichEdit; const URLText: string; Button: TMouseButton);
begin
  dxShellExecute(Sender.Handle, URLText);
end;

procedure TWebServiceDemoSetupWizard.teChange(
  Sender: TObject);
begin
  UpdateStateButtons;
end;

procedure TWebServiceDemoSetupWizard.UpdateStateButtons;
begin
  btnStart.Enabled := ((Trim(teMSGraphClientID.Text) <> '') and (Trim(teMSGraphClientSecret.Text) <> '')) or
    ((Trim(teGoogleApiClientID.Text) <> '') and (Trim(teGoogleApiClientSecret.Text) <> ''))
end;

end.
